bigtable-grpc-interface/src/main/java/com/google/cloud/bigtable/config/CredentialOptions.java

```diff
@@ -0,0 +1,100 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.config;
+
+import com.google.common.base.Preconditions;
+
+public class CredentialOptions {
+  public static final String SERVICE_ACCOUNT_JSON_ENV_VARIABLE = "GOOGLE_APPLICATION_CREDENTIALS";
+  protected static final Logger LOG = new Logger(CredentialOptions.class);
+
+  public enum CredentialType {
+    Json, MetadataServiceAccount, None, P12,
+  }
+
+  /**
+   * Credential options to get the credentials from a Json file set in the
+   * GOOGLE_APPLICATION_CREDENTIALS environment variable -
+   * https://developers.google.com/identity/protocols/application-default-credentials
+   */
+  public static CredentialOptions jsonCredential() {
+    return new CredentialOptions(CredentialType.Json);
+  }
+
+  public static boolean hasJsonCredentialConfigured() {
+    return System.getenv().containsKey(SERVICE_ACCOUNT_JSON_ENV_VARIABLE);
+  }
+
+  public static CredentialOptions metadataServiceAccountCredential() {
+    return new CredentialOptions(CredentialType.MetadataServiceAccount);
+  }
+
+  public static CredentialOptions noCredential() {
+    LOG.info("Enabling the use of null credentials. This should not be used in production.");
+    return new CredentialOptions(CredentialType.None);
+  }
+
+  public static CredentialOptions p12Credential(String serviceAccount, String keyFile) {
+    return new CredentialOptions(serviceAccount, keyFile);
+  }
+
+  private final CredentialType credentialType;
+  private final String serviceAccount;
+  private final String keyFile;
+
+  private CredentialOptions(CredentialType credentialType) {
+    this.credentialType = credentialType;
+    this.serviceAccount = null;
+    this.keyFile = null;
+  }
+
+  private CredentialOptions(String serviceAccount, String keyFile) {
+    this.credentialType = CredentialType.P12;
+    this.serviceAccount = serviceAccount;
+    this.keyFile = keyFile;
+  }
+
+  public CredentialType getCredentialType() {
+    return credentialType;
+  }
+
+  /**
+   * ServiceAccount email address used for the P12 file -
+   * https://developers.google.com/identity/protocols/OAuth2ServiceAccount
+   */
+  public String getServiceAccount() {
+    Preconditions.checkArgument(credentialType == CredentialType.P12,
+      "Only P12 credentials have a service account");
+    return serviceAccount;
+  }
+
+  /**
+   * P12 file -
+   * https://developers.google.com/identity/protocols/OAuth2ServiceAccount
+   */
+  public String getKeyFile() {
+    Preconditions.checkArgument(credentialType == CredentialType.P12,
+        "Only P12 credentials have a key file");
+
+    return keyFile;
+  }
+
+  public String getJsonFile() {
+    Preconditions.checkArgument(credentialType == CredentialType.Json,
+        "Only Json credentials have a key file");
```

Comment:
"Only P12 credentials have a key file" and "Only Json credentials have a key file"?

When we set CredentialType:
if (credentialType == CredentialType.P12) {
   Preconditions.checkArgument(hasJsonCredentialConfigured());
}